bot/monkey_patches.py

```diff
@@ -50,3 +52,36 @@ async def honeybadger_type(self, channel_id: int) -> None:  # noqa: ANN001
             pass
 
     http.HTTPClient.send_typing = honeybadger_type
+
+
+def patch_message_converter() -> None:
+    """
+    Make the Message converter infer channelID from the given context.
+
+    Discord.py's Message converter is supposed to infer channelID based
+    on ctx.channel if only a messageID is given. A "refactor" (linked below)
+    a few weeks before d.py's archival broke this, so that if only a messageID
+    is given to the converter, it will only find that message if it's in the bot's
+    cache.
+
+    https://github.com/Rapptz/discord.py/commit/1a4e73d59932cdbe7bf2c281f25e32529fc7ae1f
+    """
+    log.debug("Patching Message converter to infer channelID from context.")
+    original = commands.PartialMessageConverter._get_id_matches
+
+    def _get_as_snowflake(data: t.Any, key: str) -> t.Optional[int]:
+        try:
+            value = data[key]
+        except KeyError:
+            return None
+        else:
+            return value and int(value)
+
+    def patch(ctx: commands.Context, argument: str) -> t.Callable:
+        """Infer channel ID from context if not explicitly specified."""
+        with unittest.mock.patch("discord.utils._get_as_snowflake") as mock:
```

Comment:
its possible to patch with a provided object, rather than patching with a mock.

we can provide the new= param, https://docs.python.org/3/library/unittest.mock.html?highlight=unittest%20mock#patch

This means that we don't need to use a mock at all, which should save making a useless objects (and all of its internal workings, unittest isn't exactly known for being fast in prod)

Reply:
I don't think using mocks here is really a problem. This function is only called when something explicitly calls the message converter, and I struggled to find anything that does to test this function with. Mind you that the message object in context, and the message passed to `on_message` do not go through this. The only thing that goes through this are commands which accept a message as their arg.

Reply:
The above method would result in a performance boost from the current code. I feel like mocks are a bit slower than other objects, given their internals.

Reply:
You're talking performance boosts that will show benefits if we're calling this a 1000 times a second, but that's simply never going to be the case. We may as well do it, but honestly, it should be fine even if we don't.